lib/resources/aws/aws_config_delivery_channel.rb

```diff
@@ -28,29 +28,29 @@ def validate_params(raw_params)
       allowed_scalar_type: String,
     )
 
-    # Make sure channel_name is given as param
-    if validated_params[:channel_name].nil?
-      raise ArgumentError, 'You must provide a channel_name to aws_config_delivery_channel'
-    end
-
     validated_params
   end
 
   def fetch_from_api
     backend = BackendFactory.create(inspec_runner)
-    query = { delivery_channel_names: [@channel_name] }
-    catch_aws_errors do
-      @resp = backend.describe_delivery_channels(query)
+    query = @channel_name ? { delivery_channel_names: [@channel_name] } : {}
+    response = {}
+    begin
+      response = backend.describe_delivery_channels(query)
+    rescue Aws::ConfigService::Errors::NoSuchDeliveryChannelException
```

Comment:
Dropping this rescue to the bottom of the method would clear up the "happy path" for readers and removes the need to early return when handling the exception.

```ruby
def fetch_from_api
  # ...
rescue Aws::ConfigService::Errors::NoSuchDeliveryChannelException
  @exists = false
end
```